relayer/src/link/pending.rs

```diff
@@ -177,16 +178,28 @@ impl<Chain: ChainHandle> PendingTxs<Chain> {
                         // relayer to resubmit the transaction to the chain again.
                         error!("timed out while confirming {}", tx_hashes);
 
-                        let resubmit_res = resubmit(pending.original_od.clone());
-
-                        match resubmit_res {
-                            Ok(reply) => {
-                                self.insert_new_pending_tx(reply, pending.original_od);
-                                Ok(None)
+                        match resubmit {
+                            Some(f) => {
+                                let new_od = relay_path
+                                    .regenerate_operational_data(pending.original_od.clone());
+                                match new_od.map(f) {
+                                    Some(Ok(reply)) => {
+                                        self.insert_new_pending_tx(reply, pending.original_od);
+                                        Ok(None)
+                                    }
+                                    Some(Err(e)) => {
+                                        self.pending_queue.push_back(pending);
+                                        Err(e)
+                                    }
+                                    None => {
+                                        // No operational data was regenerated; nothing to resubmit
+                                        Ok(None)
+                                    }
+                                }
                             }
-                            Err(e) => {
-                                self.pending_queue.push_back(pending);
-                                Err(e)
+                            None => {
+                                // The clear packet interval is 0 such that no resubmit logic was received
```

Comment:
Maybe we should also document this assumption (i.e., when clear packet interval is 0 then we expect no resubmit logic was received) in the method doc above?